src/main/java/org/moqucu/games/nightstalker/sprite/object/RobotBullet.java

```diff
@@ -0,0 +1,142 @@
+package org.moqucu.games.nightstalker.sprite.object;
+
+import javafx.geometry.Point2D;
+import javafx.scene.image.Image;
+import lombok.*;
+import lombok.extern.log4j.Log4j2;
+import org.moqucu.games.nightstalker.model.Direction;
+import org.moqucu.games.nightstalker.sprite.ArtificiallyMovableSprite;
+import org.moqucu.games.nightstalker.sprite.Collidable;
+import org.springframework.statemachine.StateContext;
+import org.springframework.statemachine.StateMachine;
+import org.springframework.statemachine.config.StateMachineBuilder;
+import org.springframework.statemachine.listener.StateMachineListenerAdapter;
+
+import java.util.EnumSet;
+import java.util.Map;
+
+import static org.moqucu.games.nightstalker.NightStalkerRevived.translate;
+
+@Data
+@SuppressWarnings("unused")
+@Log4j2
+@EqualsAndHashCode(callSuper = true)
+public class RobotBullet extends ArtificiallyMovableSprite implements Collidable {
```

Comment:
How much overlap is there with the other bullet? Good-object oriented programming means to combine all the same states (= variables) and behavior (= methods) in a superclass and then extend it for the specific needs.